MapboxDirections/MBRoadClasses.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+
+public typealias RoadClasses = MBRoadClasses
+
+extension RoadClasses: CustomStringConvertible {
+    
+    public init?(descriptions: [String]) {
+        var roadClasses: RoadClasses = []
+        for description in descriptions {
+            switch description {
+            case "toll":
+                roadClasses.insert(.toll)
+            case "restricted":
+                roadClasses.insert(.restricted)
+            case "motorway":
+                roadClasses.insert(.motorway)
+            case "ferry":
+                roadClasses.insert(.ferry)
+            case "none":
+                break
+            default:
+                return nil
+            }
+        }
+        self.init(rawValue: roadClasses.rawValue)
+    }
+    
+    public var description: String {
+        if isEmpty {
+            return "none"
```

Comment:
It should be possible to round-trip a road class from an array of strings to a RoadClasses and back to a string. Is `none` a valid string that would appear in a route response? If not, maybe an empty string would be better. Unfortunately, our hands our kind of tied because CustomStringConvertible doesn’t allow for an optional description.